### ISSUE TITLE
Unable to start REPL: Unknown error

### ISSUE THREAD

**Issue by enragedginger** [cursive-ide/cursive#1437]
Hi,

Recently, I lost the ability to launch a Cursive Clojure REPL from Intellij. This formerly worked up until recently and I'm not sure what change on my system caused this.

Attempting to launch a REPL from a project.clj produces this message:
`Error running REPL for your-project-name-goes-here: Unknown error`

Mac OS X 10.11.5
IntelliJ IDEA 2016.1.3
Build #IC-145.1617, built on June 3, 2016
JRE: 1.8.0_76-release-b198 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

I'm able to reproduce this behavior by using either Cursive 1.2.7-2016.1 as well as whatever the current EAP build is.

I first noticed this while trying to continue work on a Leiningen project which formerly did not produce this behavior. However, just to rule out any project specific issues, I created a brand new Leiningen project using the app template and ran into the same error. Other things I've tried to no avail:
- `lein clean`
- Removing Intellij project files (both the .iml file and the .idea directory)
- Adding .nrepl-port and .lein-repl-port files
- Upgrading the project JDK from 1.8.0_76 to 1.8.0_91

Here's what does work:
- Running `lein repl` from the command line
- Using "Use clojure.main in normal JVM process" for the launch / run configuration instead of the default "Run nREPL with Leiningen" or "Use nREPL in normal JVM process" options.

Here's the error I'm seeing in the logs:
```
2016-06-16 08:05:39,493 [ 848578]   WARN - xecution.runners.ExecutionUtil - Execution error without description
com.intellij.execution.ExecutionException
        at cursive.repl.process_runner$create_run_profile_state$reify__12001.createJavaParameters(process_runner.clj:366)
        at com.intellij.execution.configurations.JavaCommandLineState.getJavaParameters(JavaCommandLineState.java:34)
        at cursive.repl.runner.LeiningenReplRunner$2.run(LeiningenReplRunner.java:146)
        at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:563)
        at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:142)
        at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:446)
        at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
        at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
        at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:127)
        at com.intellij.openapi.progress.impl.ProgressManagerImpl$1.run(ProgressManagerImpl.java:126)
        at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:369)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
        at cursive.repl.server$local_port.invoke(server.clj:15)
        at cursive.repl.process_runner$create_lein_parameters$reify__11985.invoke(process_runner.clj:318)
        at cursive.leiningen.project.LeinShimProvider.withShim(ProjectDetails.kt:50)
        at cursive.repl.process_runner$create_lein_parameters.invoke(process_runner.clj:309)
```

**Comment by enragedginger** [comment]
Is there anything else I can do to help troubleshoot / fix this?

**Comment by cursive-ide** [comment]
I just looked at this. It looks like your local REPL server hasn't started correctly, which nREPL then uses for acking when it's started up. Can you get your log file from *Help&rarr;Show log in Finder* and either attach it here or email it to cursive@cursive-ide.com?

**Comment by enragedginger** [comment]
I've sent in the logs. After looking at them a bit more closely, I did find this stacktrace too:
```
ERROR -   cursive.config.InitComponent - Error initialising: 
java.net.BindException: Can't assign requested address
	at java.net.PlainSocketImpl.socketBind(Native Method)
	at java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:387)
	at java.net.ServerSocket.bind(ServerSocket.java:375)
	at java.net.ServerSocket.bind(ServerSocket.java:329)
	at clojure.tools.nrepl.server$start_server.doInvoke(server.clj:143)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at cursive.repl.server$start_server.invoke(server.clj:19)
	at cursive.initialise$initialise_all.invoke(initialise.clj:36)
	at clojure.lang.Var.invoke(Var.java:375)
	at cursive.config.InitComponent.initClojure(InitComponent.java:122)
	at cursive.config.InitComponent.initComponent(InitComponent.java:78)
	at com.intellij.openapi.components.impl.ComponentManagerImpl$ComponentConfigComponentAdapter.getComponentInstance(ComponentManagerImpl.java:522)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.createComponents(ComponentManagerImpl.java:123)
	at com.intellij.openapi.application.impl.ApplicationImpl.access$801(ApplicationImpl.java:92)
	at com.intellij.openapi.application.impl.ApplicationImpl$12.run(ApplicationImpl.java:500)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:142)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:127)
	at com.intellij.openapi.application.impl.ApplicationImpl.createComponents(ApplicationImpl.java:516)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.init(ComponentManagerImpl.java:107)
	at com.intellij.openapi.application.impl.ApplicationImpl.load(ApplicationImpl.java:458)
	at com.intellij.openapi.application.impl.ApplicationImpl.load(ApplicationImpl.java:450)
	at com.intellij.idea.IdeaApplication.run(IdeaApplication.java:194)
	at com.intellij.idea.MainImpl$1$1$1.run(MainImpl.java:52)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:366)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

I'm guessing that's somehow related. For now, I've just switched over to starting a Leiningen REPL from the terminal, and then connecting to it via a remote Clojure REPL Intellij run configuration.

**Comment by cursive-ide** [comment]
Yes, that's the issue - that means that the internal REPL server in Cursive couldn't start, which the nREPL support uses to pass the port it starts on back to Cursive. I've never seen that error, but a quick google suggests it's some sort of misconfiguration on your machine. Mostly they talk about a service already listening on that port, but in this case unless you've set a special system property Cursive binds to port 0 so that's not the issue. The most likely candidate is that you have localhost misconfigured in `/etc/hosts`, or something to do with your IP changing.

I'm pretty sure this isn't a Cursive issue so I'll close this. Let me know if you need more help or if you can't figure it out.

**Comment by enragedginger** [comment]
It looks like this was indeed an error with my /etc/hosts file. I had added an alias for a docker container I was running which looked something like this:
`192.168.99.100 name-of-my-docker-image localhost`

The `localhost` bit was a vestigial remain of a one week stint with Docker Native. I removed that part (so the entry became `192.168.99.100 name-of-my-docker-image`), restarted Intellij, and then everything worked again.

Thank you!
